### DISCUSSION TITLE
Will this work if integrated as a GUI function?

### DISCUSSION THREAD

**Discussion by Morsmalleo** [nexe/nexe#995] [Q&A]
G'day guys, just wondering if *nexe* would be able work as an electron GUI function to build a NodeJS executable from source?? Say something like a NodeJS Socket.io client.

For example, let's say we have the source code for our NodeJS Socket.io client App, stored in a directory like the following;
```shell
$ MyApp/MyApp-Server/app/app/source/exe
```
and we want to compile it to a *Windows Executable(.exe)* via the click of a button on our GUI, would this be possible??

**Comment by calebboyd** [comment]
For sure! I don't see why this wouldn't work. I'd give it a try. You should be able to use the programatic API.

**Reply by Morsmalleo** [reply]
Awesome!! 😃 As for the programmatic API I assume you mean the NodeJS API I read about on the README.md or am I mistaken??
